exams/utils.py

```diff
@@ -79,44 +79,151 @@ def authorize_for_exam(mmtrack, course_run):
-            ).exists()
+    # If exam is not set on selected course then we dont need to process authorization
+    if not course_run.has_exam:
+        errors_message = message_not_eligible_template.format(
+            user=mmtrack.user.username,
+            course_id=course_run.edx_course_key
+        )
+        raise ExamAuthorizationException(errors_message)
+
+    # If user has not paid for course then we dont need to process authorization
+    if not mmtrack.has_paid(course_run.edx_course_key):
+        errors_message = message_not_eligible_template.format(
+            user=mmtrack.user.username,
+            course_id=course_run.edx_course_key
+        )
+        raise ExamAuthorizationException(errors_message)
+
+    now = datetime.datetime.now(tz=pytz.UTC)
+    # if user paid for a course then create his exam profile if it is not creaated yet.
+    ExamProfile.objects.get_or_create(profile=mmtrack.user.profile)
+
+    # if user passed the course and currently not authorization for that run then give
+    # her authorizations.
+    ok_for_authorization = (
+        mmtrack.has_passed_course(course_run.edx_course_key) and
+        not ExamAuthorization.objects.filter(
+            user=mmtrack.user,
+            course=course_run.course,
+            date_first_eligible__lte=now,
+            date_last_eligible__gte=now
+        ).exists()
+    )
+
+    if not ok_for_authorization:
+        errors_message = message_not_passed_or_exist_template.format(
+            user=mmtrack.user.username,
+            course_id=course_run.edx_course_key
         )
+        raise ExamAuthorizationException(errors_message)
 
-        if ok_for_authorization:
-            ExamAuthorization.objects.create(
-                user=mmtrack.user,
-                course=course_run.course,
-                date_first_eligible=now,
-                date_last_eligible=add_year(now)
+    ExamAuthorization.objects.create(
+        user=mmtrack.user,
+        course=course_run.course,
+        date_first_eligible=now,
+        date_last_eligible=add_year(now)
+    )
+    log.info(
+        '[Exam authorization] user "%s" is authorize for exam the for course id "%s"',
+        mmtrack.user.username,
+        course_run.edx_course_key
+    )
+
+
+def bulk_authorize_for_exam(program_id=None, username=None):
+    """
+    Authorize user(s) for exam(s).
+
+    Args:
+        program_id(str): Program id (optional) if you want authorization on specific program exams
+        username(str): User name (optional) whom you want to authorize for exam(s)
+    """
+    programs = Program.objects.filter(
+        ~Q(exam_series_code__exact=""),
+        live=True,
+        exam_series_code__isnull=False
+    )
+    if program_id is not None:
+        programs = programs.filter(id=program_id)
+
+    if not programs.exists():
+        if program_id is not None:
+            raise ExamAuthorizationException('[Exam authorization] Invalid program id: %s', program_id)
```

Comment:
I did a test without setting the `program.exam_series_code` and the management command correctly raises an exception at this point, but the message is confusing because the program is right, it is the `exam_series_code` that is missing.
Can you please update this error message saying that either the `program id` is wrong or the `exam_series_code` is missing?